.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8u171-jdk-node-browsers
+    resource_class: xlarge
+    environment:
+      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
+      _JAVA_OPTIONS: "-Xmx4g -XX:+HeapDumpOnOutOfMemoryError -verbose:gc -XX:+PrintGC -XX:+PrintGCDateStamps -XX:+PrintGCDetails -XX:-TraceClassUnloading -Xloggc:build-%t-%p.gc.log"
+      TERM: dumb
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+      - restore_cache:
+          key: gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}
+
+      - run: ./gradlew --parallel resolveConfigurations
+
+      - save_cache:
+          key: gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+          paths:
+            - ~/.gradle/wrapper
+      - save_cache:
+          key: gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}
+          paths:
+            - ~/.gradle/caches
+      - run: git tag -l | xargs git tag -d && git fetch -t # ensure all tags are fetched and up-to-date
+      - run: ./scripts/circle-ci/download-moreutils.sh
+      - run: ./scripts/time-cmd.sh ./gradlew --profile --parallel --stacktrace classes testClasses
+      - persist_to_workspace:
+          root: .
+          paths: [ . ]
+      - run: scripts/circle-ci/pr-changelog-status-check.sh
+
+  test:
+    parallelism: 8
+    docker:
+      - image: circleci/openjdk:8u171-jdk-node-browsers
+    resource_class: xlarge
+    environment:
+      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
+      _JAVA_OPTIONS: "-Xmx4g -XX:+HeapDumpOnOutOfMemoryError -verbose:gc -XX:+PrintGC -XX:+PrintGCDateStamps -XX:+PrintGCDetails -XX:-TraceClassUnloading -Xloggc:build-%t-%p.gc.log"
+      TERM: dumb
+      CIRCLE_TEST_REPORTS: /tmp/test-reports
+      CIRCLE_ARTIFACTS: /tmp/artifacts
```

Comment:
If possible, I'd recommend using exactly the same values that we use internally for this:

```
# Setup default locations for Circle test reports + artifacts
ENV CIRCLE_TEST_REPORTS=/home/circleci/junit CIRCLE_ARTIFACTS=/home/circleci/artifacts
RUN mkdir -p /home/circleci/junit && mkdir -p /home/circleci/artifacts
```

Reply:
Yep, will standardise on this. I don't think we depend on the directories being in any specific place